### ISSUE TITLE
Per Player view distance is broken from 1.9 and forward

### ISSUE THREAD

**Issue by tlaundal** [PaperMC/Paper#344]
It seems the per player view distance API implementation is broken from 1.9 and forward. I've verified this with a simple plugin, as described [here](https://aquifermc.org/threads/using-the-per-player-view-distance-api.197/).
### What behaviour is expected:

When `Player.setViewDistance(int)` the view distance for that player should be updated. Meaning more chunks should be sent to the Player if the view distance is increased, and the appropriate number of chunks should be sent when the player moves to unloaded places.
### What behaviour is observed:

If `Player.setViewDistance(int)` is called with an argument lower than the view distance currently set in that world, nothing happens.
If `Player.setViewDistance(int)` is called with an argument higher than the view distance currently set in that world, new chunks will not load for that player.
### Steps/models to reproduce:

See the plugin described [here](https://aquifermc.org/threads/using-the-per-player-view-distance-api.197/).
### Plugin list:

Plugin mentioned above
### Paper build number:

Verified on 773 and 788. Works fine on 443

**Comment by tlaundal** [comment]
Some digging revealed the reason for this bug.

The methods `PlayerChunkMap.removePlayer(...)` and `PlayerChunkMap.addPlayer(...)` needs to iterate over the chunks that should be sent to the player. On build 443 both had their loops changed to this:

```
// Paper start - Player view distance API
        int viewDistance = entityplayer.getViewDistance();
        for (int k = i - viewDistance; k <= i + viewDistance; ++k) {
            for (int l = j - viewDistance; l <= j + viewDistance; ++l) {
                // Paper end
```

However, from version 1.9 (only tested 1.9.4 to be honest), that change is only applied to `PlayerChunkMap.removePlayer(...)`. Applying the exact same change to  `PlayerChunkMap.addPlayer(...)` seems to fix the issue.

I have not pushed a fix for this because I currently don't have the time and I'm not familiar with the patch system used here. However, I tried my hand at the 1.9.4 source, which is the version I'm using, and everything worked fine then with the fix described above.

**Comment by zachbr** [comment]
Fixed in 4d7917a87da61465f54c597ff53d41bedaa27026, thanks

**Comment by tlaundal** [comment]
Thanks!
